logstash-core/lib/logstash/codecs/base.rb

```diff
@@ -11,7 +11,7 @@ module LogStash::Codecs; class Base < LogStash::Plugin
 
   def initialize(params={})
     super
-    config_init(@params)
+    config_init(@allow_env, @params)
```

Comment:
haha, that is the same question I had for `@params`, where is that made?

The answer is `plugin.rb`. when we call `super`, it does all this logic